### PR TITLE
Implement parametric tests for bootstrap

### PR DESCRIPTION
**Pull request by litchipi**
Add more tests for `massa-bootstrap`
Adds everything required to perform parametric testing easily
Test edge cases using parametric testing
Based on top of https://github.com/massalabs/massa/pull/4453

### CODE REVIEW

**Review by Leo-Besancon — COMMENTED** [review]
Nice test improvements!
A few general comments:
- Can you look at https://github.com/massalabs/massa/pull/4447 for overlaps / conflicts with your changes on final_state
- Do your tests include ser / deser at the limits (on top of randomized tests, we could manually test with data on the limit / just over the limit to see that the checks are well made)
- Another thing to uint test is the behaviour of  a bootstrap server (on what conditions it can fail / panic / ...)

File: massa-bootstrap/src/messages.rs [hunk 1]
```diff
@@ -762,7 +826,7 @@ impl BootstrapClientMessageDeserializer {
     ) -> Self {
         Self {
             id_deserializer: U32VarIntDeserializer::new(Included(0), Included(u32::MAX)),
-            length_error_deserializer: U32VarIntDeserializer::new(Included(0), Included(100000)),
+            length_error_deserializer: U32VarIntDeserializer::new(Included(0), Included(u32::MAX)),
```
[the remark this answers is not in the document]
Comment (litchipi) [reply massa-bootstrap/src/messages.rs:775]: Because in the serialization, no max length is given, this is to ensure that the behavior is the same on both sides.
If you want, we can move this value to a new `MAX_ERROR_LENGTH` constant, and check it at serialization side also

**Review by AurelienFT — APPROVED** [review]
